src/PrestaShopBundle/Routing/LegacyRouterChecker.php

```diff
@@ -116,6 +120,13 @@ public function check(Request $request): bool
 
             $controllerClass = $controllers[strtolower($queryController)];
         }
+        // Loading controller
+        $adminController = new $controllerClass();
+        $adminController->init();
+
+        $request->attributes->set(self::LEGACY_CONTROLLER_INSTANCE_ATTRIBUTE, $adminController);
```

Comment:
I'm hesitating here, should we have multiple attributes on the root level? Or one `_legacy_controller` array attribute with sub fields?

Not that it changes much things anyway, just a thought (maybe not the best one 😅)